### PR TITLE
Feature/2031 replace the old topics table with the material react table new

### PR DESCRIPTION
**Pull request by HrisyToteva**
closes https://github.com/dfds/cloudplatform/issues/2031

# Additional Review Notes
image
 In the original table the drop-down column is on the right side of the table. It should be doable in the new table as well by adding this `positionExpandColumn="last"` which I did but it doesn't work. I can think of other solutions if you don't like it on the left side.

### CODE REVIEW

File: src/src/pages/topics/SearchView.js [hunk 1]
```diff
@@ -94,6 +94,7 @@ export function SearchView({ data, onTopicClicked }) {
     }
 
     async function fetchData(data) {
+      console.log(data);
```
Comment (SEQUOIIA) [inline src/src/pages/topics/SearchView.js:97]: Be gone!

File: src/src/pages/topics/rowDetails.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    const [contracts, setContracts] = useState([]);
+    const [selectedMessageContractId, setSelectedMessageContractId] = useState(null);
+    const { selfServiceApiClient } = useContext(AppContext);
+
+    useEffect(() => {
+        fetchData(data);
+    }, []);
+
+    async function fetchData(topic) {
+        console.log(topic);
```
Comment (SEQUOIIA) [inline src/src/pages/topics/rowDetails.js:20]: Be gone!

File: src/src/pages/topics/SearchView.js (hunk 1 above)
Comment (andreasfrisch) [inline src/src/pages/topics/SearchView.js:97]: Let me help here 
```suggestion

```

File: src/src/pages/topics/rowDetails.js (hunk 2 above)
Comment (andreasfrisch) [inline src/src/pages/topics/rowDetails.js:20]: Let me help here 
```suggestion

```

**Review by andreasfrisch — COMMENTED** [review]
Just helped out Emil, has not read the whole thing yet

File: src/src/pages/topics/index.js [hunk 3]
```diff
@@ -197,9 +200,26 @@ function Topics() {
...
+        header: 'arrow',
+        id: 'arrow',
+        size: '1',
+        enableColumnFilterModes: false,
+        muiTableBodyCellProps: {
+          align: 'center',
+        },
+        Cell: ({ cell}) => {
+          return <div>
+            {!cell.row.getIsExpanded() ? ( <ChevronDown /> ) : <ChevronUp />}
```
Comment (kralle333) [inline src/src/pages/topics/index.js:219]: I would reverse the logic something like: `cell.row.getIsExpanded() ? <ChevronUp />  : <ChevronDown />`

File: src/src/pages/topics/index.js [hunk 4]
```diff
@@ -216,8 +236,8 @@ function Topics() {
         ) : (
           <>
             <MaterialReactTable columns={columns} data={filteredData}
-              positionExpandColumn="last"
-              displayColumnDefOptions={{                
+              positionExpandColumn="left"
```
Comment (kralle333) [inline src/src/pages/topics/index.js:239]: Add a comment that this is a hack :) Is there a better way to turn off this feature?

Comment (HrisyToteva) [reply src/src/pages/topics/index.js]: Actually adding the initial state of ordering columns fixed that for us. 

```
 initialState={{
                columnOrder: [
                  'name',
                  'arrow',
                ],
              }}
```

**Review by kralle333 — APPROVED** [review]
Amazing! Good job!

**Pull request merged**